internal/tiger/cmd/db.go

```diff
@@ -932,3 +960,363 @@ func isConnectionRejected(err error) bool {
+			return m, tea.Quit
+		case "2":
+			m.cursor = 1
+			m.selected = optionResetPassword
+			m.done = true
+			return m, tea.Quit
+		case "3":
+			m.cursor = 2
+			m.selected = optionExit
+			m.done = true
+			return m, tea.Quit
+		}
+	}
+	return m, nil
+}
+
+func (m passwordRecoveryModel) View() string {
+	s := "Authentication failed. What would you like to do?\n\n"
+
+	for i, option := range m.options {
+		cursor := " "
+		if m.cursor == i {
+			cursor = ">"
+		}
+		s += fmt.Sprintf("%s %d. %s\n", cursor, i+1, option)
+	}
+
+	s += "\nUse ↑/↓ arrows or number keys to select, enter to confirm, q to quit"
+	return s
+}
+
+// selectPasswordRecoveryOption shows the interactive menu for password recovery
+func selectPasswordRecoveryOption(out io.Writer) (passwordRecoveryOption, error) {
+	model := newPasswordRecoveryModel()
+
+	program := tea.NewProgram(model, tea.WithOutput(out))
+	finalModel, err := program.Run()
+	if err != nil {
+		return optionExit, fmt.Errorf("failed to run password recovery menu: %w", err)
+	}
+
+	result := finalModel.(passwordRecoveryModel)
+	return result.selected, nil
+}
+
+// updateAndSaveServicePassword updates a service password via API and saves it locally.
+// It handles the API call and password storage.
+func updateAndSaveServicePassword(
+	ctx context.Context,
+	client api.ClientWithResponsesInterface,
+	projectID string,
+	service api.Service,
+	newPassword string,
+	role string,
+	statusOut io.Writer,
+) error {
+	serviceID := *service.ServiceId
+
+	// Call API to update password
+	updateReq := api.UpdatePasswordInput{Password: newPassword}
+	resp, err := client.PostProjectsProjectIdServicesServiceIdUpdatePasswordWithResponse(ctx, projectID, serviceID, updateReq)
+	if err != nil {
+		return fmt.Errorf("failed to update password: %w", err)
+	}
+
+	if resp.StatusCode() != 200 && resp.StatusCode() != 204 {
+		return common.ExitWithErrorFromStatusCode(resp.StatusCode(), resp.JSON4XX)
+	}
+
+	// Save password locally
+	result, err := common.SavePasswordWithResult(service, newPassword, role)
+	if err != nil {
+		fmt.Fprintf(statusOut, "Warning: could not save password: %v\n", err)
+	} else if result.Success {
+		fmt.Fprintf(statusOut, "%s\n", result.Message)
+	}
+
+	return nil
+}
+
+// resetServicePassword generates a new password and updates it via API
+func resetServicePassword(ctx context.Context, cfg *config.Config, service api.Service, role string, statusOut io.Writer) (string, error) {
+	// Generate new password
+	newPassword, err := generateSecurePassword(32)
+	if err != nil {
+		return "", fmt.Errorf("failed to generate new password: %w", err)
+	}
+
+	// Get API credentials
+	apiKey, projectID, err := getCredentialsForService()
+	if err != nil {
+		return "", fmt.Errorf("authentication required: %w. Please run 'tiger auth login'", err)
+	}
+
+	// Create API client
+	client, err := api.NewTigerClient(cfg, apiKey)
+	if err != nil {
+		return "", fmt.Errorf("failed to create API client: %w", err)
+	}
```

Comment:
Similar to my comment about loading the config multiple times, the `getServiceDetails` function already [loads the credentials and API client](https://github.com/timescale/tiger-cli/blob/b936b4a160a1914f50feb9b2ff96179f942a694d/internal/tiger/cmd/db.go#L794-L804) too. Imo, we should load them once in `RunE` and pass them down everywhere they're needed.